app/indexers/hyrax/valkyrie_indexer.rb

```diff
@@ -83,5 +72,30 @@ def to_solr
         "updated_at_dtsi": resource.updated_at
       }
     end
+
+    class << self
+      ##
+      # @api public
+      # @param resource [Valkyrie::Resource] an instance of a Valkyrie::Resource or an inherited class
+      # @note This will attempt to return an indexer following a naming convention where the indexer for a resource
+      #       class is expected to be the class name appended with 'Indexer'.  It will return default ValkyrieIndexer
+      #       if an indexer following the naming convention is not found or the resource is the reserved Hyrax::Resource
+      #       which should use ValkyrieIndexer instead of the base implementation in Hyrax::ResourceIndexer.
+      # @example
+      #     resource class:  Book
+      #     indexer class:   BookIndexer
+      # @return [Valkyrie::Indexer] an instance of ValkyrieIndexer or an inherited class based on naming convention
+      #
+      def for(resource:)
+        indexer_class = resource.class.name == 'Hyrax::Resource' ? ValkyrieIndexer : indexer_from_classname(resource)
```

Comment:
it's not my favorite situation, but i think it's where we're at. 

TY!

Reply:
I didn't think it would be your favorite.  Always better to avoid hardcoding exceptions to the rule.  I tried to compensate by adding clear method documentation.